src/persist-client/src/operators/shard_source.rs

```diff
@@ -263,22 +262,48 @@ where
             return;
         }
 
+        let client = clients
+            .open(location)
+            .await
```

Comment:
I didn't do this because these async calls are normally expected to return fast.

This is the comment explaining why we do it for `subscribe()`: https://github.com/MaterializeInc/materialize/blob/5709b9a4e11e33cbc01e29bc8b919762f95d687c/src/persist-client/src/operators/shard_source.rs#L287

Is that still ok for you?

Reply:
I think if it's easy, we should do it. otherwise, there's some risk that someone comes in later and adds a call here that isn't expected to return so quickly (because why wouldn't that be okay) and introduces a hard-to-debug race. if it's not easy/adds a lot of complexity, i'd like to at least have a comment nearby to make future code modifiers aware of the issue

Reply:
I pushed a commit that adds this. It's very boilerplate-y and I'm not sure I like it. 😅